File: src/main/java/io/github/bot/terminal/application/domains/common/notify/NotifyRestConverter.java
package io.github.bot.terminal.application.domains.common.notify;


import io.github.bot.terminal.application.domains.common.notify.dto.NotifyDTO;
import io.github.bot.terminal.application.domains.common.notify.entity.NotifyDetails;
import io.github.bot.terminal.application.domains.common.notify.requests.NotifyRequest;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;


@Service
@RequiredArgsConstructor
public class NotifyRestConverter {

    public NotifyDetails mapToDetails(NotifyRequest request) {
        return new NotifyDetails()
                .setMessageTemplate(request.getMessageTemplate())
                .setNotificationApiId(request.getNotificationApiId());
    }

    public NotifyDTO mapToDto(NotifyDetails details) {
        return new NotifyDTO()
                .setMessageTemplate(details.getMessageTemplate())
                .setNotificationApiId(details.getNotificationApiId());
    }
}

File: src/main/kotlin/io/github/bot/terminal/application/domains/common/notify/NotifyRestConverter.kt
package io.github.bot.terminal.application.domains.common.notify

import io.github.bot.terminal.application.domains.common.notify.dto.NotifyDTO
import io.github.bot.terminal.application.domains.common.notify.entity.NotifyDetails
import io.github.bot.terminal.application.domains.common.notify.requests.NotifyRequest
import org.springframework.stereotype.Service

@Service
class NotifyRestConverter {

    fun mapToDetails(request: NotifyRequest): NotifyDetails =
            NotifyDetails(
                    messageTemplate = request.messageTemplate,
                    notificationApiId = request.notificationApiId
            )

    fun mapToDto(details: NotifyDetails): NotifyDTO =
            NotifyDTO(
                    messageTemplate = details.messageTemplate,
                    notificationApiId = details.notificationApiId
            )
}